File: Resources/Archive-2016/demos-news/iOSApps101Demo/iOSApps101Demo/MainViewController.m
//
//  ViewController.m
//  iOSApps101Demo
//
//  Created by Doncho Minkov on 1/22/16.
//  Copyright © 2016 Doncho Minkov. All rights reserved.
//

#import "MainViewController.h"

@interface MainViewController ()

@property (weak, nonatomic) IBOutlet UITableView *datesViews;
@property NSMutableArray *dates;

@end

@implementation MainViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.datesViews.dataSource = self;
    self.dates = [NSMutableArray array];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)updateDateTouch:(id)sender {
    NSDate *date = [[NSDate alloc] init];
    [self.dates addObject:date];
    [self.datesViews reloadData];
}
- (UITableViewCell *)tableView:(UITableView *)tableView
         cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    static NSString *simpleTableIdentifier = @"SimpleTableItem";
    UITableViewCell *cell = [self.datesViews dequeueReusableCellWithIdentifier:simpleTableIdentifier];
    if(cell == nil) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:simpleTableIdentifier];
    }
    
    cell.textLabel.text = [NSString stringWithFormat:@"%@", self.dates[indexPath.row]];
    return cell;
}

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.dates.count;
}
@end

File: demos/iOS101Application/iOS101Application/MainViewController.swift
//
//  ViewController.swift
//  iOS101Application
//
//  Created by Doncho Minkov on 3/13/17.
//  Copyright © 2017 Doncho Minkov. All rights reserved.
//

import UIKit
import Swifter

class MainViewController: UIViewController {
    @IBOutlet weak var label: UILabel!
    @IBOutlet weak var button: UIButton!
    @IBAction func click(_ sender: Any) {
        let text = "Date: \(Date())";
        
        self.label.text = text;
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        
        let server = HttpServer()
        server["/hello"] = { .OK(.Html("You asked for " + $0.url)) }
        server.start()
        
        
        label.text = "I am Loaded!";
        
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

